src/commands/trace/trace.ts

```diff
@@ -0,0 +1,144 @@
+import chalk from 'chalk'
+import {spawnSync} from 'child_process'
+import {Command} from 'clipanion'
+import crypto from 'crypto'
+import os from 'os'
+import {parseTags} from '../../helpers/tags'
+import {apiConstructor} from './api'
+import {APIHelper, CIRCLECI, Provider, SUPPORTED_PROVIDERS} from './interfaces'
+
+// We use 127 as exit code for invalid commands since that is what *sh terminals
+const badCommandExitCode = 127
```

Comment:
I'd probably rename this to `BAD_COMMAND_EXIT_CODE`. It is more conventional

Reply:
and probably export it to be used for tests

Reply:
Done